accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+```xml
+<!-- WorkloadManifest.targets (entry point): -->
+
+<Project>
+    <Import Project="Xamarin.Manifest.Android.Framework.targets" Condition="'$(TargetPlatformIdentifier)'=='android'" />
+    <Import Project="Xamarin.Manifest.iOS.Framework.targets" Condition="'$(TargetPlatformIdentifier)'=='ios'" />
+</Project>
+
+<!-- Xamarin.Manifest.Android.Framework.targets -->
+
+<Project>
+    <Import Project="Sdk.targets" Sdk="Xamarin.Android.Sdk" />
+    <Import Project="Sdk.targets" Sdk="Xamarin.Android.Aot.Sdk" Condition="'$(UseAot)'=='true'" />
+    <!--
+    Import any other sdks based on project state.
+    -->
+</Project>
+```
+
+In this example, the iOS and Android workloads both share the same workload manifest.  In addition to `WorkloadManifest.targets`, the manifest package includes separate targets files specific to iOS and Android.  The more complex logic for which workload packs to import from is guarded behind a simple `TargetPlatformIdentifier` check in the `WorkloadManifest.targets`.
+
+### Side-by-side SDK support for new OS releases
+
+Optional workloads may version semi-independently from the .NET SDK.  Workload owners may want to support side-by-side tooling for new OS versions, in order to reduce risk of introducing regressions in a patch level change.
+
+The multiplexing between different versions of an OS SDK can be implemented in the `WorkloadManifest.targets` file that is part of the workload manifest, or other .targets it imports.  For example, the `Xamarin.Manifest.iOS.Framework.targets` file imported by `WorkloadManifest.targets` in the previous example could look like this:
+
+```xml
+<Project>
+    <Import Project="Sdk.targets" Sdk="Xamarin.iOS.Sdk" Condition="$([MSBuild]::VersionLessThan($(TargetPlatformVersion), '15.0'))" />
+    <Import Project="Sdk.targets" Sdk="Xamarin.iOS.Sdk.15.0" Condition="$([MSBuild]::VersionGreaterThanOrEquals($(TargetPlatformVersion), '15.0'))" />
+</Project>
+```
+
+This would import the iOS targets from the Xamarin.iOS.Sdk.15.0 workload pack if the `TargetPlatformVersion` is 15.0 or higher, and from the Xamarin.iOS.Sdk workload pack otherwise.
+
+### Workload props files
+
+Workloads may also provide MSBuild logic in an `AutoImport.props` file in the Sdk folder of the workload pack which will be imported before the body of the project file.  We expect this to be use for default globs, for example to specify that all .xaml files should be included as `Page` items.
+
+However, `AutoImport.props` files from workloads are subject to restrictions and must be carefully authored.  This is because since they need to be imported before the body of the project is evaluated, they can't be included conditionally based on the properties in the project that define whether the workload is in use.  So the `AutoImport.props` files from all workload packs will be imported for all projects using Microsoft.NET.Sdk.
```

Comment:
```suggestion
However, `AutoImport.props` files from workloads are subject to restrictions and must be carefully authored.  This is because since they need to be imported before the body of the project is evaluated, they can't be included conditionally based on the properties in the project that define whether the workload is in use.  So the `AutoImport.props` files from all workload packs will be imported for all projects using `Microsoft.NET.Sdk`.
```